internal/backend/runtime/omni/controllers/omni/machine_status.go

```diff
@@ -417,10 +417,14 @@ func (ctrl *MachineStatusController) handleNotification(ctx context.Context, r c
 
 		if event.Hostname != nil {
 			spec.Network.Hostname = *event.Hostname
+		} else {
+			spec.Network.Hostname = ""
```

Comment:
this I believe is not exactly great change, as by the design some parts of the event might be `nil` if they're missing.

The proper way probably is to treat missing `HostnameStatus` as a way to set `event.Hostname` to `&("")`

Reply:
Yeah. I guess I have an idea